vms/platformvm/block/proposal_block_test.go

```diff
@@ -26,17 +27,68 @@ func TestNewBanffProposalBlock(t *testing.T) {
 		parentID,
 		height,
 		proposalTx,
+		[]*txs.Tx{},
 	)
 	require.NoError(err)
 
-	// Make sure the block and tx are initialized
 	require.NotEmpty(blk.Bytes())
-	require.NotEmpty(blk.Tx.Bytes())
-	require.NotEqual(ids.Empty, blk.Tx.ID())
-	require.Equal(proposalTx.Bytes(), blk.Tx.Bytes())
+	require.Equal(parentID, blk.Parent())
+	require.Equal(height, blk.Height())
 	require.Equal(timestamp, blk.Timestamp())
+
+	blkTxs := blk.Txs()
+	require.Len(blkTxs, 1)
+	l := len(blk.Transactions)
+	expectedTxs := make([]*txs.Tx, l+1)
+	copy(expectedTxs, blk.Transactions)
+	expectedTxs[l] = blk.Tx
+	require.Equal(expectedTxs, []*txs.Tx{proposalTx})
+	for i, blkTx := range blkTxs {
+		expectedTx := expectedTxs[i]
+		require.NotEmpty(blkTx.Bytes())
+		require.NotEqual(ids.Empty, blkTx.ID())
+		require.Equal(expectedTx.Bytes(), blkTx.Bytes())
+	}
+}
+
+func TestNewBanffProposalBlockWithDecisionTxs(t *testing.T) {
```

Comment:
Do we need a new test? Why not just put `decisionTxs` in `blk` in `TestNewBanffProposalBlock`?

Reply:
I put a new test since BanffProposalBlocks with no `decisionTxs` should be valid- not sure if this is the right way to do that but open to suggestions

Reply:
Done, good suggestion 🫡